tests/unit/features/DenseFeatures_unittest.cc

```diff
@@ -119,3 +119,33 @@ TEST(DenseFeaturesTest, copy_dimension_subset_with_subsets)
 	SG_UNREF(features);
 	SG_UNREF(f_reduced);
 }
+
+TEST(DenseFeaturesTest, shallow_copy_subset_data)
```

Comment:
Need more tests.
Pls add some tests where you ensure that the objects can be used within threads. 
I.e. modifying the shallow_subset copy doesnt change another one...
etc